src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -194,10 +223,8 @@ internal static unsafe partial class Utf16Utility
                             //
                             // A UTF-16 high/low surrogate code unit has the bit pattern [ 11011q## ######## ],
                             // where # is any bit; q = 0 represents a high surrogate, and q = 1 represents
-                            // a low surrogate. Since we added 0xA800 in the vectorized operation above,
-                            // our surrogate pairs will now have the bit pattern [ 10000q## ######## ].
-                            // If we logical right-shift each word by 3, we'll end up with the bit pattern
-                            // [ 00010000 q####### ], which means that we can immediately use pmovmskb to
+                            // a low surrogate. Right-shifting each surrogate char by 3 bits, we end up with
+                            // [ 00011011 q####### ], which means that we can immediately use pmovmskb to
```

Comment:
Nitpicking: `pmovmskb` this has code paths for ARM too, so should this be reflected in the comment?

Note: besides that the comment is clear enough to know what is meant.